### PR TITLE
Elf range

### PR DESCRIPTION
**Pull request by Tachiorz**
Subj. Decided to implement 3rd combat pack type, which is elf range.

### CODE REVIEW

**Comment by exectails** [comment]
Can't merge, conflicts.

**Comment by Tachiorz** [comment]
rebased

File: src/ChannelServer/Skills/Combat/RangedAttack.cs [hunk 1]
```diff
@@ -141,103 +144,132 @@ public CombatSkillResult Use(Creature attacker, Skill skill, long targetEntityId
...
+				aAction.Stun = attackerStun;
+				cap.Add(aAction);
 
-				// More damage with fire arrow
-				if (attacker.Temp.FireArrow)
-					damage *= FireBonus;
+				// Hit by chance
+				if (rnd < chance)
+				{
+					var targetSkill = target.Skills.ActiveSkill != null ? target.Skills.ActiveSkill.Info.Id : SkillId.CombatMastery;
```
Comment (exectails) [inline src/ChannelServer/Skills/Combat/RangedAttack.cs]: Here you broke the training, told you that it depends on that the target action's skill being set to the attacking skill =/

Comment (exectails) [inline src/ChannelServer/Skills/Combat/RangedAttack.cs]: I wonder... if you hit someone who has a random skill loaded, say WM, or Smash, is the target skill id that skill or is it CombatMastery? Is that field determined *after* potentially canceling active skills or before?

**Pull request merged**